File: create_tables/ba_crew.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table ba_crew';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute(
    """create table ba_crew(
        CREW_COMPANY_BA_ID VARCHAR(40) NOT NULL,
        CREW_ID VARCHAR(40) NOT NULL,
        ACTIVE_IND VARCHAR(1),
        COST_PER_UNIT DECIMAL(12,2),
        COST_PER_UNIT_CURRENCY_UOM_UOM VARCHAR(40),
        COST_PER_UNIT_UOM VARCHAR(40),
        CREW_ABBREVIATION VARCHAR(12),
        CREW_LONG_NAME VARCHAR(255),
        CREW_SHORT_NAME VARCHAR(30),
        CREW_TYPE VARCHAR(40),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        PPDM_GUID VARCHAR(38),
        REMARK VARCHAR(2000),
        SOURCE VARCHAR(40),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")

File: postgres_39/struct/BaCrew.go
package dto

import (
	"time"
)

type Ba_crew struct {
	Crew_company_ba_id             string     `json:"crew_company_ba_id"`
	Crew_id                        string     `json:"crew_id"`
	Active_ind                     *string    `json:"active_ind"`
	Cost_per_unit                  *float64   `json:"cost_per_unit"`
	Cost_per_unit_currency_uom_uom *string    `json:"cost_per_unit_currency_uom_uom"`
	Cost_per_unit_uom              *string    `json:"cost_per_unit_uom"`
	Crew_abbreviation              *string    `json:"crew_abbreviation"`
	Crew_long_name                 *string    `json:"crew_long_name"`
	Crew_short_name                *string    `json:"crew_short_name"`
	Crew_type                      *string    `json:"crew_type"`
	Effective_date                 *time.Time `json:"effective_date"`
	Expiry_date                    *time.Time `json:"expiry_date"`
	Ppdm_guid                      string     `json:"ppdm_guid"`
	Remark                         *string    `json:"remark"`
	Source                         *string    `json:"source"`
	Row_changed_by                 *string    `json:"row_changed_by"`
	Row_changed_date               *time.Time `json:"row_changed_date"`
	Row_created_by                 *string    `json:"row_created_by"`
	Row_created_date               *time.Time `json:"row_created_date"`
	Row_effective_date             *time.Time `json:"row_effective_date"`
	Row_expiry_date                *time.Time `json:"row_expiry_date"`
	Row_quality                    *string    `json:"row_quality"`
}
